.ci/packaging.groovy

```diff
@@ -227,6 +231,24 @@ def tagAndPush(name){
         log(level: 'WARN', text: "${name} doesn't have ${variant} docker images. See https://github.com/elastic/beats/pull/21621")
       }
     }
+
+    if (aliasVersion != "") {
+      def aliasName = "${DOCKER_REGISTRY}/observability-ci/${name}${variant}:${aliasVersion}"
+      def iterations = 0
+      retryWithSleep(retries: 3, seconds: 5, backoff: true) {
+        iterations++
+        def status = sh(label:'Change tag and push', script: """
+          docker tag ${oldName} ${aliasName}
+          docker push ${newName}
+        """, returnStatus: true)
```

Comment:
Would it be possible to extract this string script in groovy to a variable, reusing https://github.com/elastic/beats/pull/21960/files#diff-5edb5ab93fc95960129eecb2ee3d8763a9c5e02b56e5b80126e0c5a52296ee8fR221-R226? Something like what we did in the Go side (https://github.com/elastic/beats/pull/21960/files#diff-2d000e355fe01e44a412b7e21904c617cc0953e8da569f66608d40068c08f163R196-R202)?

Reply:
what do you mean?

Reply:
Yes, being able to compose the string in a dynamically way:

```
// pseudocode
def stringCommand = `
docker tag ${oldName} ${newName}
docker push ${newName}
docker tag ${oldName} ${commitName}
docker push ${commitName}
`
if (aliasVersion == "") {
      stringCommand += `
          docker tag ${oldName} ${aliasName}
          docker push ${aliasName}
      `
}
```
and then pass the `$stringCommand` to the `sh` step

Reply:
Aha, if so, then the retry will even try with previous commands that already went through successfully. So, I think it should be fine with the current approach

Reply:
Wait you can do something like:

```
// supported image flavours
  def variants = ["", "-oss", "-ubi8"]
  variants.each { variant ->
    def oldName = "${DOCKER_REGISTRY}/beats/${name}${variant}:${libbetaVer}"
    def newName = "${DOCKER_REGISTRY}/observability-ci/${name}${variant}:${tagName}"
    def commitName = "${DOCKER_REGISTRY}/observability-ci/${name}${variant}:${env.GIT_BASE_COMMIT}"

    tagWithRetry(oldName, newName, name, variant)
    tagWithRetry(oldName, commitName, name, variant)

    if (aliasVersion != "") {
      def aliasName = "${DOCKER_REGISTRY}/observability-ci/${name}${variant}:${aliasVersion}"
      tagWithRetry(oldName, aliasName, name, variant)
    }
  }
}

def tagWithRetry(oldName, newName, name, variant) {
  def iterations = 0
  retryWithSleep(retries: 3, seconds: 5, backoff: true) {
    iterations++
    def status = sh(label: "Change tag and push ${newName}", script: """
      docker tag ${oldName} ${newName}
      docker push ${newName}
    """, returnStatus: true)

    if ( status > 0 && iterations < 3) {
      error('tag and push failed, retry')
    } else if ( status > 0 ) {
      log(level: 'WARN', text: "${name} doesn't have ${variant} docker images. See https://github.com/elastic/beats/pull/21621")
    }
  }
}
```

Reply:
Ok got it. I was thinking in an auxiliary method too, but thought the variable was going to be simpler 👍

Reply:
I sent this code, a little refactored here: https://github.com/elastic/beats/pull/21960/commits/d2a3f393e2beaebb562e5535953ef94f5c73b2c7